### DISCUSSION TITLE
KDE Plasma 5.27 + Wayland Can Run on OPi5+ with Ubuntu 22.04 v1.XX

### DISCUSSION THREAD

**Discussion by traxap** [Joshua-Riek/ubuntu-rockchip#248] [Q&A] [answered]
Is it possible to run KDE Plasma-Wayland? 
I installed KDE Full with the intent to run Plasma (Wayland) on OPi5+ with Ubuntu 22.04 Desktop v 1.19. Unfortunately, if I select Plasma Wayland, I am not able to login after reboot.  At the login screen it allows me to enter the password --> enter, and after a few seconds it brings me back to the login screen.  I can login if I select Plasma X11, however, any apps I open cannot be moved or resized, and the top menu bar is missing, therefore, it cannot be moved with the mouse or other keyboard combos.  
Any suggestions would be really appreciated.
Thank you.

**Comment by Joshua-Riek** [comment]
Unfornatually, Gnome is the only desktop environment with a mature wayland implementation. x11 is a buggy mess on this platform due to the current GPU driver situation.

**Comment by traxap** [comment]
Thanks for the quick reply.   Just to let you know, I really like your version of the OS.  You've done a great job with it.  It's very responsive and it works really well on the OPi5+. Thank you.

P.S.  Any ETA on the "GPU driver situation?"  Do you believe is it going to be fixed, eventually?

**Reply by Joshua-Riek** [reply]
Thanks, I'm glad to see people enjoying it. 

As for the GPU driver situation, it's a little complex. Currently, we are using a hacked and reverse-engineered GPU driver called panfork created by icecream95. His work is impressive, and he deserves much credit. However, this is a hack. I and other developers are waiting for new Linux Kernel releases to come out that provide better support for the hardware. This new kernel will allow us to use PanCSF as an official GPU driver for this device. But, all of this is highly experimental. Most everything will need to be done by open-source developers. I hate to release something experimental like this. It would put me way out of my depth when people report issues. Then again, I'm curious and might do it anyway.  

TLDR: I guess it will take a year from now for a stable user experience and everything to work properly.

**Comment by traxap** [comment]
Sorry for being a bit late with my reply...

I am so grateful to developers like you who dedicate their time and effort to make these many flavors of Linux OSs available to the community.  I am new to Linux and my first  good experience was the Raspbian OS on Raspberry pi 4B 8gb version. The OS was a bit slow, but stable and easy to use, which allowed me to learn about the OS and Linux applications that, to my surprise, can help me jump from Windows to Linux.  Further more, with the release of the more powerful RockChip's SBCs, it made it possible for myself and others to build little systems that are affordable and powerful enough to compete with the big boys.  Once the system was up and running, and stable, (...your flavor of Ubuntu), I was able to direct my attention to the large number of Linux applications... e.g. currently I am working in LibreOffice in the same way I work in MSOffice. This is totally amazing!!!  

...Looking forward to see the GPU implementation in your version of Ubuntu...  As I mentioned earlier, your version of the OS is stable, responsive and easy to use, and it let's me work rather than fix and maintain.  Once I get the KDE Plasma to work on my OPi5+ system it will look and feel amazing. No reasons to go back to Windows, and I look forward to that.  

I am learning a lot and I am excited about the future of Linux!
Thank you.

P.S. 
Let me know if I can help with any testing. 
My system: OPi5+ 16GB, 1TB NMVe Booting/Working Drive

**Comment by cristianapas** [comment]
I was able to run KDE Plasma 5.24.7 by installing the plasma-workspace-wayland package, soon I will try to install version 5.27 using PPA

PD: Sorry for my english :)

**Reply by cristianapas** [reply, marked as answer]
Definitely version 5.27 works much better, to install it use the experimental PPA and then upgrade using pkcon since only with apt upgrade it doesn't work.

`sudo add-apt-repository ppa:kubuntu-ppa/experimental`
`sudo apt update`
`sudo pkcon update`

**Reply by traxap** [reply]
Thank you for letting me know.  I'll try it and let you know.

**Reply by traxap** [reply]
Thank you "cristianapas" for posting these steps. With your help I was able to finally get KDE Plasma 5.27 installed and running on my OP5+.

First I opened a terminal and did the following;

sudo apt update
sudo apt upgrade

Next I followed your instructions.

Installed:
plasma-workspace-wayland  - with Synaptic Package Manager

Opened a terminal and ran these commands.

sudo add-apt-repository ppa:kubuntu-ppa/experimental
sudo apt update
sudo pkcon update

After I executed the above commands I proceeded with KDE Plasma installation using Tasksel.
You can also install the full KDE Plasma using this command:  sudo apt install kde-full 

Using the same terminal you can install KDE via Tasksel with this command:  

sudo apt update && sudo upgrade 
sudo apt-get install tasksel 

-- and select SDDM during the process.

Next I typed "tasksel" in the terminal and hit the Enter key.

When the Software selection option screen appeared I selected KDE Plasma and selected OK, and let it install it.

Next I reboot/restart the system.

Then at the Log-In page I selected Plasma (Wayland) and entered the password and I was in.

It's a bit glitchy, but responsive and quite fun to play and work in it.  Occasionally it freezes when opening apps but Log Off and Log In usually fixes it. 

Thank you again.

**Reply by hakotani-o** [reply]
If you have time, please edit /etc/profile.d/qt.sh to "export QT_QPA_PLATFORM=xcb;wayland" and try installing again.

**Reply by traxap** [reply]
Hello hakotani-o,
I am not very knowledgeable when it comes to in depth coding.  I accessed the file you mentioned and this is what I found: "export QT_QPA_PLATFORM=wayland" the line did not have the "xcb;

I edited the line you mentioned to show "export QT_QPA_PLATFORM=xcb;wayland"

![image](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/138725531/0b33e697-f48e-41e4-940e-296a63d5e208)

You mentioned about reinstalling.  What is the command line I need to us to reinstall?

Thank you for your help.

**Reply by hakotani-o** [reply]
Sorry,please try edit and reboot.
Thank you.

**Reply by traxap** [reply]
Thank you.

**Reply by hakotani-o** [reply]
Sorry again, please add single quotation.
export QT_QPA_PLATFORM='xcb;wayland'

**Reply by traxap** [reply]
I made the change, however, I've noticed the boot-up time from the selection sign-in screen to the desktop has increased from 2-3 seconds to 10-15 seconds.  Is this an expected side effect of the change?
Thanks.

**Reply by hakotani-o** [reply]
Sorry,i also thought it would take a long time to log in.
I'm building an lxqt-desktop environment, and I'm running into a problem where I can't log in.
As a countermeasure for this, I changed the QT_QPA_PLATFORM settings.
First of all, you can't do anything if you can't log in.
Unfortunately, that's all I know.
I'm also a Linux newbie.
I think Mr. Joshua Riek is a great developer.
He is providing us with the development environment for the RK3588 board.
Please let's study them one by one.

Thank you.

**Reply by traxap** [reply]
Thanks hakotani-o for your help.  I will continue to experiment as well. I'll keep you posted.

**Reply by hakotani-o** [reply]
Hello, I tried various things and finally the login became faster.
However, I don't know what worked for you, but you can try it if you like.

1. sudo systemctl status sddm
   Please check if it is started with plasma wayland.
  If it is X11, please select plasma wayland when logging in (Not now).
 /etc/profile.d/qt.sh edit
export QT_QPA_PLATFORM=wayland
2. reinstall sddm and plasma-workspace-wayland
3. reboot
4.login and change wallpaper
5. reboot

that's all

If reboot or poweroff is slow, please enter Ctrl+Alt+F1.

good luck

**Reply by jsiddall** [reply]
I got KDE 5.27 installed.  Not sure why but I also need to add the backport-extra PPA as described here: https://kubuntu.org/news/plasma-5-27-lts-for-jammy-22-04-lts-available-via-ppa/

It definitely runs better than earlier versions of KDE, but still has issues with plasma shell hanging (often when toasts are open -- hmmm...) and strange screen artifacts in various places.  It is bad enough that I added a script on a hotkey to restart it:

```
killall plasmashell
kstart5 plasmashell &
```

but obviously that is not ideal.

Firefox installed from a package also hangs a lot, doing things like viewing PDFs, and won't play any youtube videos without the tab crashing.  Chromium works better.

Tried both Wayland and X11 (with and without compositing) and all had similar behavior.
